closure/testing/test/BUILD

```diff
@@ -111,6 +111,40 @@ closure_js_test(
     ],
 )
 
+# TEST: closure_js_library targets from exports should be provided when used in
+# a data attr.
+
+closure_js_library(
+    name = "external",
+    srcs = ["external.js"],
```

Comment:
I think this part is little bit misleading. You are not trying to compile the code. That's why I recommended putting it on the data.

Reply:
In my use case, I have a closure_js_library for each of my third party libraries, and they are compiled together using forgiving settings to create a vendor bundle. However, code that requires those libraries as externs needs to include them as data for their tests. I can change this test if you want, but then it wouldn't test my actual usage of this feature.

Reply:
> In my use case, I have a closure_js_library for each of my third party libraries, and they are compiled together using forgiving settings to create a vendor bundle. However, code that requires those libraries as externs needs to include them as data for their tests.

In that care you probably want a closure_js_binary and depend that as data in the test.

> I can change this test if you want, but then it wouldn't test my actual usage of this feature.

Your use case sounds non-trivial but the test should reflect basic underlying functionality/bug for maintenance purposes.

Reply:
> In my use case, I have a closure_js_library for each of my third party libraries, and they are compiled together using forgiving settings to create a vendor bundle. However, code that requires those libraries as externs needs to include them as data for their tests.

In that care you probably want a closure_js_binary and depend that as data in the test.

> I can change this test if you want, but then it wouldn't test my actual usage of this feature.

Your use case sounds non-trivial but the test should reflect basic underlying functionality/bug for maintenance purposes.

Reply:
> In my use case, I have a closure_js_library for each of my third party libraries, and they are compiled together using forgiving settings to create a vendor bundle. However, code that requires those libraries as externs needs to include them as data for their tests.

In that care you probably want a closure_js_binary and depend that as data in the test.

> I can change this test if you want, but then it wouldn't test my actual usage of this feature.

Your use case sounds non-trivial but the test should reflect basic underlying functionality/bug for maintenance purposes.

Reply:
Yeah, those are still built with plovr, I suppose eventually we would use the binaries instead. I'll update it to just use data.

Reply:
Having a data-only closure_js_library target is not valid:

```
closure_js_library(
    name = "external",
    data = ["external.js"],
)
```

results in
```
Either 'srcs' or 'exports' must be specified
```